### ISSUE TITLE
Use usage="rec" to indicate recommended attributes

### ISSUE THREAD

**Issue by pe-ro** [music-encoding/music-encoding#320]
Use of `@usage="rec"` and display of the attribute's status in the Data Dictionary will help clarify when an attribute is recommended. Since "rec" results in an optional attribute in RNG (the same as "opt"), there is no compatibility issue. Need to compile a list of recommended attributes.

**Comment by pe-ro** [comment]
Since compiling a list of recommended attributes for every element is likely to take quite some time, I'm kicking this issue further down the road to someday.

**Comment by bwbohl** [comment]
Here’s a deduplicated list of `attDef[@usage='opt']` from `source/modules`:
- a
- abbr
- aboveorder
- absolute
- abstype
- accid
- accid.ges
- accidlower
- accidlower.ges
- accidupper
- accidupper.ges
- adlib
- agent
- align
- altrend
- altsym
- amount
- analog
- angle.optimize
- angled
- arrow
- arrow.color
- arrow.fillcolor
- arrow.shape
- arrow.size
- artic
- artic.ges
- atleast
- atmost
- attacca
- attach
- attached.by
- audience
- auth
- auth.uri
- azimuth
- b
- bar.len
- bar.method
- bar.place
- bar.thru
- beam
- beam.color
- beam.group
- beam.rend
- beam.rests
- beam.slope
- beam.with
- beams
- beams.float
- beatdef
- begin
- beloworder
- betweenorder
- betype
- bezier
- block
- bracket.place
- bracket.visible
- breaksec
- bulge
- c
- calendar
- cancelaccid
- cause
- cautionary
- cert
- character
- chordref
- class
- clef.color
- clef.dis
- clef.dis.place
- clef.line
- clef.shape
- clef.visible
- closed
- cluster
- codedval
- coll
- color
- colored
- cols
- colspan
- compressable
- comptype
- con
- confidence
- contemporary
- context
- control
- copyof
- corresp
- corrlevel
- count
- cue
- currency
- curve
- curvedir
- cutout
- d
- data
- decls
- def
- deg
- degree
- delayed
- delim
- desc
- dir
- dir.dist
- dis
- dis.place
- divisio
- dot
- dots
- dots.ges
- dur
- dur.default
- dur.ges
- dur.metrical
- dur.ppq
- dur.real
- dur.recip
- dynam.dist
- e
- elevation
- enclose
- end
- enddate
- endho
- endid
- ending.rend
- endsym
- endsym.size
- endto
- endvo
- evaluate
- evidence
- expan
- expand
- extender
- extent
- extremis
- f
- facs
- fermata
- flag.form
- flag.pos
- float.gap
- folium
- follows
- fontfam
- fontname
- fontsize
- fontstyle
- fontweight
- form
- fret
- from
- func
- g
- gliss
- glyph.auth
- glyph.name
- glyph.num
- glyph.uri
- grace
- grace.time
- grid.show
- halign
- hand
- harm.dist
- head.altsym
- head.auth
- head.color
- head.fill
- head.fillcolor
- head.mod
- head.rotation
- head.shape
- head.visible
- height
- ho
- hooked
- initial
- inner.recto
- inner.verso
- instant
- instr
- interval
- inth
- intm
- inttype
- isodate
- isodur
- join
- key.accid
- key.mode
- key.pname
- keysig
- keysig.cancelaccid
- keysig.visible
- label
- lang.main
- lang.other
- layer
- layerscheme
- left
- len
- lendsym
- lendsym.size
- letterspacing
- level
- lform
- lig
- ligated
- line
- lineheight
- lines
- lines.color
- lines.visible
- loc
- long
- looped
- lrx
- lry
- lsegs
- lstartsym
- lstartsym.size
- lv
- lwidth
- lyric.align
- lyric.fam
- lyric.name
- lyric.size
- lyric.style
- lyric.weight
- material
- max
- medium
- meiversion
- mensur.color
- mensur.dot
- mensur.form
- mensur.loc
- mensur.orient
- mensur.sign
- mensur.size
- mensur.slash
- metcon
- meter.count
- meter.form
- meter.showchange
- meter.sym
- meter.unit
- meter.visible
- method
- mfunc
- midi.bpm
- midi.channel
- midi.duty
- midi.instrname
- midi.instrnum
- midi.mspb
- midi.pan
- midi.patchname
- midi.patchnum
- midi.port
- midi.track
- midi.volume
- mimetype
- min
- mm
- mm.dots
- mm.unit
- mnum.visible
- mode
- modusmaior
- modusminor
- multi.number
- music.name
- music.size
- n
- new
- next
- niente
- nonfiling
- notafter
- notationsubtype
- notationtype
- notbefore
- num
- num.default
- num.format
- num.place
- num.visible
- numbase
- numbase.default
- nymref
- occurs
- oct
- oct.default
- oct.ges
- old
- oloc
- onstaff
- ontheline
- opening
- opening.vertical
- optimize
- order
- ordered
- orient
- origin.endid
- origin.layer
- origin.startid
- origin.tstamp
- ornam
- outer.recto
- outer.verso
- page.botmar
- page.height
- page.leftmar
- page.panels
- page.rightmar
- page.scale
- page.topmar
- page.width
- part
- partstaff
- pclass
- pedal.style
- place
- plist
- ploc
- pname
- pname.ges
- pnum
- pos
- ppq
- precedes
- prefix
- prev
- prolatio
- proport.num
- proport.numbase
- psolfa
- quantity
- reason
- recordtype
- recto
- reh.dist
- reh.enclose
- rellen
- removed.by
- rend
- rendgrid
- resp
- restart
- rhythm
- right
- role
- rotate
- rotation
- rowspan
- ruledlines
- ruledstaves
- s-shape
- sameas
- scale
- scheme
- select
- seq
- shape
- sig
- sign
- since
- singleton
- slash
- slope
- slur
- slur.lform
- slur.lwidth
- solo
- source
- spaces
- spacing
- spacing.packexp
- spacing.packfact
- spacing.staff
- spacing.system
- staff
- startdate
- startho
- startid
- startsym
- startsym.size
- startto
- startvo
- state
- status
- stem.dir
- stem.form
- stem.len
- stem.mod
- stem.pos
- stem.sameas
- stem.visible
- stem.with
- stem.x
- stem.y
- syl
- sym
- symbol
- synch
- system.leftline
- system.leftmar
- system.rightmar
- system.topmar
- tab.align
- tab.anchorline
- tab.course
- tab.courses
- tab.fing
- tab.fret
- tab.line
- tab.pos
- tab.string
- tab.strings
- target
- targettype
- tempo.dist
- tempus
- text.fam
- text.name
- text.size
- text.style
- text.weight
- tie
- tie.lform
- tie.lwidth
- tilt
- to
- trans.diat
- trans.semi
- translit
- tstamp
- tstamp.ges
- tstamp.real
- tstamp2
- tstamp2.ges
- tstamp2.real
- tune.Hz
- tune.pname
- tune.temper
- tuning.standard
- tuplet
- type
- ulx
- uly
- unit
- unitdur
- val
- val2
- valign
- value
- vel
- version
- verso
- vgrp
- visible
- vo
- voltasym
- vu.height
- waves
- when
- width
- withid
- wordpos
- writtenlines
- writtenstaves
- x
- x2
- xlink:actuate
- xlink:role
- xlink:show
- xml:base
- xml:id
- xml:lang
- xml:space
- y
- y2

**Comment by bwbohl** [comment]
Here’s a deduplicated list of `attDef[@usage='rec']` from source/modules:
- dur.quality
- origin.staff
- origin.tstamp2
- func
- form
- place
- tilt

**Comment by bwbohl** [comment]
Here’s an example of how the difference appears in an element-spec in the Guidelines

<img width="1105" height="148" alt="Image" src="https://github.com/user-attachments/assets/4249c28a-17c8-4b9e-8873-65b5d9cac341" />

**Comment by pe-ro** [comment]
@bwbohl, thanks for moving this forward.  I can choose which elements I think are recommended, but it would be helpful to have input from others. @lpugin? @kepper?

**Comment by rettinghaus** [comment]
Thanks @bwbohl for this list. Maybe it would be helpful to see the attribute classes as well? `@func` e.g. is optional in some cases, in others it is recommended or even required (for `repeatMark`).

**Comment by bwbohl** [comment]
@pe-ro what do you think, an we close this or should this become a bigger refactoring endeavour revisting all elements and their recommended attributes? I think we have a lot of recommended attributes and the guidelines state ther recommended status so thwe initial issue might be resolve already.

**Comment by pe-ro** [comment]
Both keeping it open or closing it are fine by me. The Guidelines can (already do?) address whether an attribute is recommended (and under what conditions, as these may change as @rettinghaus points out), but it would be helpful to show this in the ODD view as well for those users that actually use this view (like me). Attribute status placed here can also provide "ground truth" against which the Guidelines can be checked.

**Comment by bwbohl** [comment]
Yes, ground truth helps. Guess we’ll keep it open for a while then ;-)
What is it you are refering to as _ODD view_?

**Comment by pe-ro** [comment]
I use the term "ODD view" to mean what one sees when selecting something other than "Guidelines" on the page at https://music-encoding.org/guidelines/v5/content/, leading to, for example, 
https://music-encoding.org/guidelines/v5/elements/abbr.html#attributes_full_tab. In other words, the stuff that is derived from the files in [/source/modules](https://github.com/music-encoding/music-encoding/tree/develop/source/modules).

When I use the term "Guidelines" I mean the textual explanations presented when the "Guidelines" link is selected; that is, material derived from the files in [/source/docs](https://github.com/music-encoding/music-encoding/tree/develop/source/docs).

I realize this may not be how everyone refers to these things.
